controllers/company/index.test.js

```diff
@@ -0,0 +1,123 @@
+const request = require('supertest')
+const app = require('../../index')
+const faker = require('faker')
+const truncate = require('../../utils/truncate')
+const factory = require('../../utils/helpers/factories')
+const { fakerCompany } = require('../../utils/helpers/fakers')
+
+const companyMock = {
+  company: {
+    name: 'Company fullname ltda 123',
+    fullname: 'Fullname company social name ltda',
+    document: String(faker.random.number()),
+    siteUrl: 'www.mycompany.com.br',
+    allowOrder: true,
+    allowPdv: false
+  },
+  user: {
+    name: 'Alexandre Soares',
+    email: `${faker.random.number()}@hotmail.com`,
+    password: '123456',
+    birthday: 'Mon Mar 15 2021 16:01:24 GMT-0300 (Horário Padrão de Brasília)'
+  }
+}
+
+describe('company controller', () => {
+  describe('post company', () => {
+    // afterAll(async () => await truncate())
+
+    it('create company', async () => {
+      expect.hasAssertions()
+      const res = await request(app).post('/register').send(companyMock)
+      expect(res.statusCode).toBe(201)
+      expect(res.request.method).toStrictEqual('POST')
+      expect(companyMock.company).toHaveProperty('name')
+      expect(companyMock.company).toHaveProperty('fullname')
+      expect(companyMock.company).toHaveProperty('document')
+      expect(companyMock.company).toHaveProperty('siteUrl')
+      expect(companyMock.company).toHaveProperty('allowOrder')
+      expect(companyMock.company).toHaveProperty('allowPdv')
+      expect(companyMock.user).toHaveProperty('name')
+      expect(companyMock.user).toHaveProperty('email')
+      expect(companyMock.user).toHaveProperty('password')
+      expect(companyMock.user).toHaveProperty('birthday')
```

Comment:
você deve fazer a vericficação da sua resposta e não dos dados que está passando.
```suggestion
      expect(res.body).toHaveProperty('name', companyMock.company.name)
```